operator/pkg/model/translation/envoy_virtual_host.go

```diff
@@ -449,7 +480,10 @@ func getRouteAction(route *model.HTTPRoute, backends []model.Backend, backendHTT
 func getRouteRedirect(redirect *model.HTTPRequestRedirectFilter, listenerPort uint32) *envoy_config_route_v3.Route_Redirect {
 	redirectAction := &envoy_config_route_v3.RedirectAction{}
 
+	// Add scheme-based condition to prevent redirect loops
 	if redirect.Scheme != nil {
+		// Add a header matcher to only apply the redirect if the scheme doesn't match the target scheme
+		// This prevents redirect loops when the request already has the target scheme
```

Comment:
This comment is incorrect; there is nothing added here. Why add this comment at all?